### PR TITLE
Neo 3.2.0/1 updates

### LINKED ISSUE #640: Update WitnessScopes
Update witness scoping according to the results of https://github.com/neo-project/neo/pull/2596.

### LINKED ISSUE #753: Add native CryptoLib.murmur32() method to devpack
See https://github.com/neo-project/neo/pull/2604

### LINKED ISSUE #754: Add native LedgerContract.getTransactionVMState() to devpack
See https://github.com/neo-project/neo/pull/2628

### PR DESCRIPTION
**Pull request by mialbu**
Closes #753
Closes #754
Closes #640

### CODE REVIEW

**Comment by gsmachado** [comment]
This will be a large one... 😅  🚀  💪

**Comment by gsmachado** [comment]
This will be a large one... 😅  🚀  💪

**Comment by mialbu** [comment]
#759 will not be included in this PR due to issues when accessing a list entry when a signer with witness scope `rules` is present. #759 will be handled separately in PR #764.

File: core/src/test/java/io/neow3j/protocol/core/witnessrule/WitnessRuleTest.java [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+public class WitnessRuleTest extends ResponseTester {
```
Comment (csmuller) [inline core/src/test/java/io/neow3j/protocol/core/witnessrule/WitnessRuleTest.java:25]: 😨 All these tests. The burden... Good work!

**Review by csmuller — CHANGES_REQUESTED** [review]
Good work!

I pushed two minor changes directly.

Can you add one integration test in `Neow3jReadOnlyIntegrationTest`or `Neow3jWriteIntegrationTest` that uses a `WitnessRule` in its witness? Just to make sure that we are really passing the correctly serialised transaction data and the neo-node can read our request.

File: core/src/main/java/io/neow3j/transaction/witnessrule/AndCondition.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        this();
+        if (expressions.length > MAX_SUBITEMS) {
+            throw new IllegalArgumentException(
+                    format("A maximum of %s subitems is allowed for an AND witness condition.", MAX_SUBITEMS));
+        }
+        this.expressions = asList(expressions);
+    }
+
+    @Override
+    public io.neow3j.protocol.core.witnessrule.WitnessCondition toJson() {
```
[the remark this answers is not in the document]
Comment (mialbu) [reply core/src/main/java/io/neow3j/transaction/witnessrule/AndCondition.java:35]: I agree. So, we both win! ^^

**Review by csmuller — APPROVED** [review]

**Pull request merged**